packages/block-editor/src/store/reducer.js

```diff
@@ -1344,13 +1344,32 @@ export function automaticChangeStatus( state, action ) {
 			return;
 		// Undoing an automatic change should still be possible after mouse
 		// move.
+		case 'SET_SELECTED_MOUNTED_BLOCK':
 		case 'STOP_TYPING':
 			return state;
 	}
 
 	// Reset the state by default (for any action not handled).
 }
 
+/**
+ * Reducer returning selected and mounted block. This state is useful for
+ * components rendering and positioning controls around the block's node.
+ *
+ * @param {boolean} state  Current state.
+ * @param {Object}  action Dispatched action.
+ *
+ * @return {boolean} Updated state.
+ */
+export function selectedMountedBlock( state, action ) {
```

Comment:
How is this different from the block selection state?

Reply:
It's only set when the block node is mounted. This is useful when you have to render a popover at the position of this node. We can't render the popover before the node is mounted.

Reply:
I honestly feel this is entirely useless. If the selected block is in the store, it's already mounted right?

Reply:
I don't think it's useless, but we can try alternative ways for sure. Being a selected block in the store doesn't mean that the block is mounted. React first has to render the block. This is similar to the block node not being available at the time of rendering. You have to use `useEffect`.

Reply:
I'm open to alternatives though.

Reply:
I feel having a state just to say "something is mounted" is not great. ideally, something in the state should make the block mount, in the case of blocks, blocks are mounted as soon as they are in the state, which for me suggests that it should be treated at the component level. 

I guess the main issue here is that the rendering of the block and the rendering of the toolbar arae separate in two separate React trees so for the first render of the blocks, we're not certain which one will render first.

I'm not sure what's the perfect solution here other than just using an interval/useEffect or something like that.

Reply:
Right, the only dependency on `BlockListBlock` is the position of the DOM node. I totally agree with you that it's not ideal and that's why I marked it unstable. I'll explore an alternative idea in a separate PR.

Reply:
We could try to provide `Popover` with a selector for the anchor. `Popover` already repositions at an interval, so this could work. Additionally, we also already use a selector for multi selected block (to enable sticking the toolbar).